crates/noirc_frontend/src/lexer/token.rs

```diff
@@ -60,80 +18,124 @@ pub enum Token {
     Keyword(Keyword),
     IntType(IntType),
     Attribute(Attribute),
-    // <
+    /// <
     Less,
-    // <=
+    /// <=
     LessEqual,
-    // >
+    /// >
     Greater,
-    // >=
+    /// >=
     GreaterEqual,
-    // ==
+    /// ==
     Equal,
-    // !=
+    /// !=
     NotEqual,
-    // +
+    /// +
     Plus,
-    // -
+    /// -
     Minus,
-    // *
+    /// *
     Star,
-    // /
+    /// /
     Slash,
-    // %
+    /// %
     Percent,
-    // &
+    /// &
     Ampersand,
-    // ^
+    /// ^
     Caret,
-    // <<
+    /// <<
     ShiftLeft,
-    // >>
+    /// >>
     ShiftRight,
-    // .
+    /// .
     Dot,
-    // ..
+    /// ..
     DoubleDot,
-    // (
+    /// (
     LeftParen,
-    // )
+    /// )
     RightParen,
-    // {
+    /// {
     LeftBrace,
-    // }
+    /// }
     RightBrace,
-    // [
+    /// [
     LeftBracket,
-    // ]
+    /// ]
     RightBracket,
-    // ->
+    /// ->
     Arrow,
-    // |
+    /// |
     Pipe,
-    // #
+    /// #
     Pound,
-    // ,
+    /// ,
     Comma,
-    // :
+    /// :
     Colon,
-    // ::
+    /// ::
     DoubleColon,
-    // ;
+    /// ;
     Semicolon,
-    // !
+    /// !
     Bang,
-    // _
+    /// _
     Underscore,
-    // =
+    /// =
     Assign,
     #[allow(clippy::upper_case_acronyms)]
     EOF,
 
-    // An invalid character is one that is not in noir's language or grammar.
-    // Delaying reporting these as errors until parsing improves error messages
+    /// An invalid character is one that is not in noir's language or grammar.
+    /// Delaying reporting these as errors until parsing improves error messages
```

Comment:
Nit: We delay report these as errors until the parser improves its error handling